### PR TITLE
[BREAKING] refactor!: do not require griefing collateral argument

### PR DESCRIPTION
**Pull request by gregdhill**
Signed-off-by: Gregory Hill <gregorydhill@outlook.com>

This PR removes the griefing collateral parameters from `request_issue` and `request_replace` since we anyway override the value and it is complicated to determine ahead-of-time.

### CODE REVIEW

**Comment by sander2** [comment]
>  since we anyway override the value
No we don't? We only check that it's at least the minimum amount. 

But I would be totally happy to remove this argument but I agree that it feels useless. We had a conversation about this a long long time ago, and if I remember correctly, Dom or Alexei wanted to keep the argument so that the user has a known upper bound for the griefing collateral. But again, I'm in favor of removing.

**Comment by gregdhill** [comment]
> > since we anyway override the value
> 
> No we don't? We only check that it's at least the minimum amount.

I meant that we lock on the given amount as opposed to the expected amount (see [here](https://github.com/interlay/interbtc/blob/2df5c5a4988452125abb31d24ac359b8fc6e3a39/crates/issue/src/lib.rs#L295-L299)). I don't see any reason why the user should lock more than is necessary so also happy to change this and just have the input parameter be an upper limit.

It's worth noting that this is a common source of confusion on Kintsugi since the input amount can be quickly invalidated.

**Comment by sander2** [comment]
> > > since we anyway override the value
> > 
> > 
> > No we don't? We only check that it's at least the minimum amount.
> 
> I meant that we lock on the given amount as opposed to the expected amount (see [here](https://github.com/interlay/interbtc/blob/2df5c5a4988452125abb31d24ac359b8fc6e3a39/crates/issue/src/lib.rs#L295-L299)). I don't see any reason why the user should lock more than is necessary so also happy to change this and just have the input parameter be an upper limit.
> 
> It's worth noting that this is a common source of confusion on Kintsugi since the input amount can be quickly invalidated.

Ah we meant the same thing, I just misunderstood what you were referring to as "the value"

**Comment by sander2** [comment]
I would be inclined to split the PR: one for the removal in issue, and one for replace. The reason being that replace is a lot more tricky: the way griefing collateral is handled there (and interacts with e.g. redeem) is a bit complicated

File: crates/replace/src/lib.rs [hunk 1]
```diff
@@ -326,19 +320,12 @@ impl<T: Config> Pallet<T> {
 
         let requestable_tokens = ext::vault_registry::requestable_to_be_replaced_tokens::<T>(&vault_id)?;
         let to_be_replaced_increase = amount_btc.min(&requestable_tokens)?;
-        let replace_collateral_increase = if amount_btc.is_zero() {
-            griefing_collateral
-        } else {
-            ext::vault_registry::calculate_collateral::<T>(&griefing_collateral, &to_be_replaced_increase, &amount_btc)?
-        };
+
+        ensure!(!amount_btc.is_zero(), Error::<T>::ReplaceAmountZero);
```
Comment (sander2) [inline crates/replace/src/lib.rs:324]:
```suggestion
        ensure!(!to_be_replaced_increase.is_zero(), Error::<T>::ReplaceAmountZero);
```
Can you also add a test for this? Fyi, I think adding zero btc was allowed because it allowed you to increase the griefing collateral, but with your change it no longer makes sense

File: crates/replace/src/lib.rs [hunk 2]
```diff
@@ -349,13 +336,10 @@ impl<T: Config> Pallet<T> {
         );
 
         // check that that the total griefing collateral is sufficient
-        let required_collateral = ext::fee::get_replace_griefing_collateral::<T>(
```
Comment (sander2) [inline crates/replace/src/lib.rs:352]: I can't comment on the line above, because you didn't change it, but this is no longer true

```
        // check that total-to-be-replaced is above the minimum. NOTE: this means that even
        // a request with amount=0 is valid, as long the _total_ is above DUST. This might
        // be the case if the vault just wants to increase the griefing collateral, for example.
   ```

File: crates/vault-registry/src/lib.rs [hunk 3]
```diff
@@ -1035,7 +1033,7 @@ impl<T: Config> Pallet<T> {
             increase: tokens.amount(),
         });
 
-        Ok((new_to_be_replaced, new_collateral))
+        Ok((new_to_be_replaced, vault.replace_collateral()))
```
Comment (sander2) [inline crates/vault-registry/src/lib.rs:1036]: It doesn't really make sense to me to return the replace collateral in a function that does nothing with the collateral. Maybe just add a getter? Also the comment above the function is outdated

File: crates/replace/src/lib.rs [hunk 4]
```diff
@@ -349,13 +336,10 @@ impl<T: Config> Pallet<T> {
...
+        let griefing_collateral = ext::fee::get_replace_griefing_collateral::<T>(
```
Comment (sander2) [inline crates/replace/src/lib.rs:339]:
```suggestion
        let griefing_collateral = ext::fee::get_replace_griefing_collateral::<T>(to_be_replaced_increase);
```
And remove `let replace_collateral_increase = griefing_collateral.saturating_sub(&previous_griefing_collateral)?;`

**Review by sander2 — CHANGES_REQUESTED** [review]
Also a note: vaults can no longer call request_replace to re-broadcast the event (now they'll have to withdraw first)

**Review by sander2 — DISMISSED** [review]

**Review by sander2 — APPROVED** [review]
